lib/rust/parser/src/source/code.rs

```diff
@@ -13,98 +13,133 @@ use crate::prelude::*;
 #[derive(Debug, Clone, Default, Eq, PartialEq, Deref)]
 pub struct StrRef<'s>(pub &'s str);
 
-/// A code representation. It can either be a borrowed source code or a modified owned one.
+/// Identifies a location in source code.
+#[derive(
+    Copy,
+    Clone,
+    Debug,
+    Default,
+    Eq,
+    PartialEq,
+    Serialize,
+    Reflect,
+    Deserialize,
+    PartialOrd,
+    Ord
+)]
+pub struct Location {
+    /// Offset from the beginning, in UTF-8 code units (bytes).
+    #[reflect(hide)]
+    pub utf8:  u32,
+    /// Offset from the beginning, in UTF-16 code units (two-byte words).
+    #[reflect(hide)]
+    pub utf16: u32,
+    /// Offset from the first line.
+    #[reflect(hide)]
+    pub line:  u32,
+    /// Offset from start of line, in Unicode characters.
+    #[reflect(hide)]
+    pub col:   u32,
```

Comment:
What does "unicode characters" mean here exactly? The amount of utf16 code units, or codepoints?

Reply:
It should be UTF16 code units. Fixed.